File: prob_09.py
def is_pyth(a,b,c):
	if a*a + b*b == c*c:
		return True
	else:
		return False

for i in range(1,1000):
	for j in range(i,1000):
		k = 1000-i-j
		if is_pyth(i,j,k):
			print i*j*k

File: Ruby/problems/prob_09.rb
class Prob9
  def self.run(n=1000)
    (1..n).to_a.each do |i|
      (i..n).to_a.each do |j|
        k = n - i - j
        return i*j*k if self.is_pyth?(i, j, k)
      end
    end
  end

  private
  def self.is_pyth?(a, b, c)
    a*a + b*b == c*c
  end
end
